src/graphql/graphql.schema.ts

```diff
@@ -3,23 +3,25 @@ const gql = (input: any) => input
 
 const schema = gql`
   union Results =
-      DatasetSearchResultType
-    | CMSSearchResultType
-    | DataSearchResultType
+      DatasetResult
+    | CMSResult
+    | MixedDataResult
     | MapLayer
     | MapCollection
+    | MixedMapResult
 
   interface SearchResult {
     totalCount: Int!
+    # TODO: See if results can be made required.
     results: [Results!]
     filters: [Filter!]
     pageInfo: PageInfo!
   }
 
-  interface SearchResultType {
+  interface MixedResult {
```

Comment:
Not sure about using `Mixed*` as naming for typings, can we rename this to something more understandable?